### PR TITLE
optional verbose nana cmake

### PR DESCRIPTION
**Pull request by qPCR4vir**
Recomended by @KingDuckZ in https://github.com/cnjinhao/nana/pull/278
Silent warnings from boost

### CODE REVIEW

**Comment by qPCR4vir** [comment]
The API file filebox.hpp already depends on filesystem for folderbox API: 
then lets filebox API take a fs::path too for init_path and init_file?? It can make it easier to check

File: source/gui/filebox.cpp [hunk 1]
```diff
@@ -1278,17 +1290,22 @@ namespace nana
...
-		brw.ulFlags = BIF_RETURNONLYFSDIRS | BIF_NEWDIALOGSTYLE;
-		brw.lpfn = browse_folder_callback;
-
+		auto title = to_wstring( impl_->title ) ;
 		std::wstring init_path = impl_->init_path.wstring();
-		brw.lParam = reinterpret_cast<LPARAM>(init_path.c_str());
-		auto pidl = ::SHBrowseForFolder(&brw);
 
+		// https://docs.microsoft.com/en-us/windows/desktop/api/shlobj_core/ns-shlobj_core-_browseinfoa
+        BROWSEINFO brw       = { 0 };
```
Comment (beru) [inline source/gui/filebox.cpp]: Here, 8 spaces are used for indentation. Using 2 tabs might be good to keep consistency.

File: include/nana/gui/filebox.hpp [hunk 2]
```diff
@@ -103,6 +103,11 @@ namespace nana
 		{
 			return show();
 		}
+
+		/// Set a new title for the dialog
+		/// @param	string	a text for title
+		/// @return the old title.
+		::std::string title( ::std::string new_title);
```
Comment (beru) [inline include/nana/gui/filebox.hpp]: The argument type should be const reference. I hope `std::string_view` will be used in future though.

File: include/nana/gui/filebox.hpp [hunk 3]
```diff
@@ -9,7 +9,7 @@
  *
  *	@file nana/gui/filebox.hpp
  *  @author Jinhao
- *  @brief a dialog to chose file(s), implemented "native" in windows but using nana for X11 
+ *  @brief dialogs to chose file(s) or a directory, implemented "native" in windows but using nana for X11
```
Comment (beru) [inline include/nana/gui/filebox.hpp]: How about splitting files for filebox and folderbox?

Comment (beru) [reply include/nana/gui/filebox.hpp]: `choose` should be used here instead of `chose`.

Comment (cnjinhao) [reply include/nana/gui/filebox.hpp]: OK

File: include/nana/gui/filebox.hpp (hunk 2 above)
Comment (cnjinhao) [reply include/nana/gui/filebox.hpp]: OK

**Comment by cnjinhao** [comment]
Thanks for your PR!

**Pull request merged**